### PR TITLE
Update lists of catalogers

### PR DESCRIPTION
**Pull request by jsquyres**
Update the lists of catalogers to be current, and sort the lists for easier human readability, both in the README and in `syft/pkg/cataloger/cataloger.go`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,39 +152,53 @@ This default behavior can be overridden with the `default-image-pull-source` con
...
+- rpm-db
 - java
+- graalvm-native-image
+- apkdb
 - go-module-binary
 - dotnet-deps
+- portage
+- nix-store
+- sbom
+- binary
```
Comment (kzantow) [inline README.md:169]: It looks like this list isn't alphabetical -- would you mind re-sorting it, too?

Comment (kzantow) [reply README.md:169]: And maybe while you're at it resort the catalogers alphabetically in: https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/cataloger.go ?

Comment (jsquyres) [reply README.md:169]: 🤦‍♂️ 

I sorted it, but apparently I goofed!  Let me fix and re-push.

**Review by kzantow — APPROVED** [review]
Thanks for the cleanup @jsquyres !

**Pull request merged**